torch/csrc/jit/tracer.cpp

```diff
@@ -343,6 +345,39 @@ void exit(const Stack& outputs) {
   setTracingState(nullptr);
 }
 
+std::pair<std::shared_ptr<TracingState>, Stack> trace(
+  Stack inputs,
+  std::function<Stack(Stack)> traced_fn,
+  std::function<std::string(const Variable&)> var_name_lookup_fn,
+  bool force_outplace,
+  script::Module* self
+) {
+  try {
+    std::shared_ptr<TracingState> state;
+    Stack trace_state_inputs;
+    std::tie(state, trace_state_inputs) = tracer::enter(std::move(inputs), self);
```

Comment:
tracer enter/exit should now be private. They should be marked static. I think they are small enough, that they can just be manually inlined here.